scripts/query-spec

```diff
@@ -9,37 +29,58 @@ if [ "${VERBOSE:-0}" -ge 2 ] || [ "${DEBUG:-0}" -eq 1 ]; then
     ERR_OUTPUT="/dev/stderr"
 fi
 
-if [ $# -lt 2 ]; then
-    echo "Usage: $0 file.spec field" >&2
+source_dir="$1"
+spec_file="$2"
+query_field="$3"
+dist="$4"
+
+if [ $# -lt 4 ]; then
+    echo "Usage: $0 source_dir file.spec field dist" >&2
     exit 1
 fi
 
-[[ "x$(basename "${1}.in")" == "x.in" ]] && exit 0
+rpm_defines=()
+
+# make it a function and export, as bash doesn't support exporting an array
+set_rpm_defines() {
+    local dist="$1"
+    # strip possible devel number
+    raw_dist="${dist#*.}"
+    if [[ "$raw_dist" = "fc"* ]]; then
+        dist_ver="${raw_dist#fc}"
+        rpm_defines+=(--define "fedora $dist_ver")
+    elif [[ "$raw_dist" = "el"* ]]; then
+        dist_ver="${raw_dist#el}"
+        rpm_defines+=(--define "centos $dist_ver")
+        rpm_defines+=(--define "rhel $dist_ver")
+    fi
+}
+
+export -f set_rpm_defines
+
+spec_file_bn="$(basename "${spec_file}")"
 
-# $RPM_QUERY_DEFINES doesn't work here for multiple reasons:
-# - it isn't exported by Makefile.rpmbuilder
-# - it contains quoted spaces, which aren't handled properly when are part of a
-#   variable
-# To workaround this, define %{dist} manually - should be good enough for most
-# queries done through this script.
+[[ "${spec_file_bn}.in" == ".in" ]] && exit 0
 
-if [ -r "${1}.in" ]; then
-    #rpm -q $RPM_QUERY_DEFINES --qf "$2" --specfile <(`dirname $0`/generate-spec "${1}.in" /dev/stdout) 2>/dev/null
+if [ -r "${spec_file}.in" ]; then
+    #rpm -q $RPM_QUERY_DEFINES --qf "$2" --specfile <(`dirname $0`/generate-spec "${spec_file}.in" /dev/stdout) 2>/dev/null
     # need to create a file due to a bug in process substitution (e.g. artwork package)
     tmp_spec=$(mktemp --tmpdir tmp.XXXXXX.spec)
-    "$(dirname "$0")/generate-spec" "${1}.in" "${tmp_spec}" 2> $ERR_OUTPUT
-    if [ "$2" = "%{SOURCE0}" ]; then
-        "$(dirname "$0")/../tools/spectool" --list-files --source 0 "${tmp_spec}" 2> $ERR_OUTPUT
+    "$(dirname "$0")/generate-spec" "$source_dir" "${spec_file}.in" "${tmp_spec}" 2>$ERR_OUTPUT
+    if [ "${query_field}" = "%{SOURCE0}" ]; then
+        spectool --list-files --source 0 "${tmp_spec}" 2>$ERR_OUTPUT
+    elif [ "${query_field}" = "PACKAGES_LIST" ]; then
+        "$(dirname "$0")"/query-builtrpms "${tmp_spec}" "$dist" 2>$ERR_OUTPUT
```

Comment:
This script is no in here. And also `PACKAGES_LIST` field is not queried in legacy builder.